dspace-server-webapp/src/test/java/org/dspace/app/rest/csv/CSVMetadataImportReferenceTest.java

```diff
@@ -0,0 +1,328 @@
+     * a UUID reference with multiable references
+     */
+    @Test
+    public void testMultiUUIDReference() throws Exception {
+        Item person = ItemBuilder.createItem(context, col1)
+                .withRelationshipType("Person")
+                .build();
+        Item person2 = ItemBuilder.createItem(context, col1)
+                .withRelationshipType("Person")
+                .build();
+        String[] csv = {"id,relationship.type,relation.isAuthorOfPublication,collection,rowName,dc.identifier.other",
+                "+,Publication," + person.getID().toString() + "||" + person2.getID().toString() + "," +
+                        col1.getHandle() + ",anything,0"};
+        Item[] items = runImport(csv);
+        assertRelationship(items[0], person, 1);
+        assertRelationship(items[0], person2, 1);
+    }
+
+    /**
+     * Test existence of newly created items with proper relationships defined in the item's metadata via
+     * multi metadata references. One archived item and one in the csv.
+     */
+    @Test
+    public void testMultiRefArchivedCsv() throws Exception {
+        Item person = ItemBuilder.createItem(context, col1)
+                .withTitle("Person")
+                .withRelationshipType("Person")
+                .build();
+        String[] csv = {"id,dc.title,relationship.type,relation.isAuthorOfPublication,collection,rowName," +
+                "dc.identifier.other",
+                "+,Person2,Person,," +  col1.getHandle() + ",idVal,0",
+                "+,Pub1,Publication,dc.title:Person||dc.title:Person2," +  col1.getHandle() + ",anything,1"};
+        Item[] items = runImport(csv);
+        assertRelationship(items[1], person, 1);
+        assertRelationship(items[1], items[0], 1);
+    }
+
+    /**
+     * Test existence of newly created items with proper relationships defined in the item's metadata via
+     * a metadata reference with a special char utilized in the metadata import script
+     */
+    @Test
+    public void testRefWithSpecialChar() throws Exception {
+        String[] csv = {"id,dc.title,relationship.type,relation.isAuthorOfPublication,collection,rowName," +
+                "dc.identifier.other",
+                "+,Person:,Person,," +  col1.getHandle() + ",idVal,0",
+                "+,Pub1,Publication,dc.title:Person:," +  col1.getHandle() + ",anything,1"};
+        Item[] items = runImport(csv);
+        assertRelationship(items[1], items[0], 1);
+    }
+
+    /**
+     * Test failure when referring to item by non unique metadata in the csv file.
+     */
+    @Test
+    public void testNonUniqueMDRefInCsv() throws Exception {
+        String[] csv = {"id,relationship.type,relation.isAuthorOfPublication,collection,dc.identifier.other",
+                "+,Person,," + col1.getHandle() + ",1",
+                "+,Person,," + col1.getHandle() + ",1",
+                "+,Publication,dc.identifier.other:1," + col1.getHandle() + ",2"};
```

Comment:
can you also verify a duplicate rowName creates an error?

Reply:
Yes, we will add a test case for two rows declaring the same rowName; this should be an early-detectable error.